nuget/lib/dependabot/nuget/update_checker/nupkg_fetcher.rb

```diff
@@ -90,31 +134,47 @@ def self.get_nuget_v3_package_url_from_search(repository_details, package_id, pa
       # rubocop:enable Metrics/PerceivedComplexity
       # rubocop:enable Metrics/CyclomaticComplexity
 
+      sig do
+        params(
+          repository_details: T::Hash[Symbol, T.untyped],
+          package_id: String,
+          package_version: String
+        )
+          .returns(T.nilable(String))
+      end
       def self.get_nuget_v2_package_url(repository_details, package_id, package_version)
         # get package XML
         base_url = repository_details[:base_url].delete_suffix("/")
         package_url = "#{base_url}/Packages(Id='#{package_id}',Version='#{package_version}')"
         response = fetch_url(package_url, repository_details)
-        return nil unless response.status == 200
+        return nil unless response&.status == 200
 
         # find relevant element
-        doc = Nokogiri::XML(response.body)
+        doc = Nokogiri::XML(T.must(response).body)
         doc.remove_namespaces!
 
         content_element = doc.xpath("/entry/content")
         nupkg_url = content_element&.attribute("src")&.value
         nupkg_url
       end
 
+      sig do
+        params(
+          stream_url: String,
+          auth_header: T::Hash[String, String],
+          max_redirects: Integer
+        )
+          .returns(T.nilable(StringIO))
+      end
       def self.fetch_stream(stream_url, auth_header, max_redirects = 5)
-        current_url = stream_url
+        current_url = T.let(stream_url, T.nilable(String))
```

Comment:
Can we avoid making this nilable  (and the T.must below) if we do T.must on `response.headers["Location"]`? I assume it's quarenteed to exist for redirect status'.

Reply:
Good point. Will do!